libs/dal/src/repositories/base-repository.ts

```diff
@@ -101,6 +110,109 @@ export class BaseRepository<T_DBModel, T_MappedEntity, T_Enforcement> {
     }
   }
 
+  private async createCursorBasedOrStatement({
+    isSortDesc,
+    paginateField,
+    after,
+    queryOrStatements,
+  }: {
+    isSortDesc: boolean;
+    paginateField?: string;
+    after: string;
+    queryOrStatements?: object[];
+  }): Promise<FilterQuery<T_DBModel>[]> {
+    const afterItem = await this.MongooseModel.findOne({ _id: after });
+    if (!afterItem) {
+      throw new DalException('Invalid after id');
+    }
+
+    let cursorOrStatements: FilterQuery<T_DBModel>[] = [];
+    let enchancedCursorOrStatements: FilterQuery<T_DBModel>[] = [];
+    if (paginateField && afterItem[paginateField]) {
+      const paginatedFieldValue = afterItem[paginateField];
+      cursorOrStatements = [
+        { [paginateField]: isSortDesc ? { $lt: paginatedFieldValue } : { $gt: paginatedFieldValue } } as any,
+        { [paginateField]: { $eq: paginatedFieldValue }, _id: isSortDesc ? { $lt: after } : { $gt: after } },
+      ];
+      const firstStatement = (queryOrStatements ?? []).map((item) => ({
+        ...item,
+        ...cursorOrStatements[0],
+      }));
+      const secondStatement = (queryOrStatements ?? []).map((item) => ({
+        ...item,
+        ...cursorOrStatements[1],
+      }));
+      enchancedCursorOrStatements = [...firstStatement, ...secondStatement];
+    } else {
+      cursorOrStatements = [{ _id: isSortDesc ? { $lt: after } : { $gt: after } }];
+      const firstStatement = (queryOrStatements ?? []).map((item) => ({
+        ...item,
+        ...cursorOrStatements[0],
+      }));
+      enchancedCursorOrStatements = [...firstStatement];
```

Comment:
similar to above but for the case when we want to paginate just by the `_id` field